### PR TITLE
[PLT-2254] Running re_dms on staging

### PR DESCRIPTION
**Pull request by joshuafleck**
[Allow updates to tables that do not exist](https://github.com/meetcleo/re_dms/commit/1032a485544c5a138a760855331b8622f1273ec6) 
Changes the behaviour from `panic` to `skip` in the case we're attempting to perform an update on a table that does not exist.

This is a legitimate use-case for situations where we want to turn on re_dms, but haven't backfilled existing data in the table.

[Bump Rust toolchain from 1.77.2 to 1.78](https://github.com/meetcleo/re_dms/commit/74ba08b254bfda59ba6a64e078c8e809d171b4a3) 
This is in advance of us migrating from Rusoto (no longer maintained) to Aws-supported Rust libraries. Most of the AWS libs depend on Rust >= 1.78

[Use IAM_ROLE in copy commands](https://github.com/meetcleo/re_dms/commit/02a5bdeb92e43863f88f9de479c1aacaf3280ad4) 
This switches auth from key-based to role-base for Redshift `copy` commands.

This paves the way to fully migrating away from IAM auth for re_dms.

### CODE REVIEW

File: roles/re_dms/files/re_dms.conf.example [hunk 1]
```diff
@@ -15,6 +15,7 @@ PG__POOL__TIMEOUTS__CREATE__NANOS=0
 PG__POOL__TIMEOUTS__RECYCLE__SECS=60
 PG__POOL__TIMEOUTS__RECYCLE__NANOS=0
 
+IAM_ROLE=
 AWS_ACCESS_KEY_ID=
```
Comment (flood4life) [inline roles/re_dms/files/re_dms.conf.example:19]: **question (non-blocking):** Do we still rely on key/secret credentials for anything? If not, shall we remove them (both from the example and eventually from the actual conf files?

File: src/database_writer.rs [hunk 2]
```diff
@@ -625,11 +618,15 @@ impl DatabaseWriter {
                 );
                 return Ok(true);
             } else if s3_file.kind == ChangeKind::Update {
-                logger_panic!(
+                // this could happen if we turn on re_dms for
+                // an existing DB, and we don't care about
+                // backfilling historical data
+                logger_error!(
```
Comment (flood4life) [inline src/database_writer.rs:624]: I wonder if this behaviour should be opt in? I feel a bit cautious allowing this by default

File: roles/re_dms/files/re_dms.conf.example (hunk 1 above)
Comment (joshuafleck) [reply roles/re_dms/files/re_dms.conf.example:19]: Yes, we require them for S3 access. https://github.com/rusoto/rusoto?tab=readme-ov-file#maintenance-status does not support `IMDSv2`, so we need to migrate to the official AWS rust library if we want to be able to get rid of these.

File: src/database_writer.rs (hunk 2 above)
Comment (joshuafleck) [reply src/database_writer.rs:624]: Sure, I can stick it behind an ENV

Comment (joshuafleck) [reply src/database_writer.rs]: Fixed in 8be9ee6cfe14c7330ee9a491172c85c19ea6f81d @flood4life

**Review by flood4life — APPROVED** [review]

**Pull request merged**